lib/Marlin/Marlin/src/module/planner.cpp

```diff
@@ -2356,7 +2356,7 @@ bool Planner::_populate_block(block_t * const block, bool split_move,
       #if IS_KINEMATIC && DISABLED(CLASSIC_JERK)
         delta_mm_cart
       #else
-        { delta_mm.x, delta_mm.y, delta_mm.z, delta_mm.e }
```

Comment:
Shouldn't we also shield this with JD_SMALL_SEGMENT_HANDLING?

Reply:
No, there is no connection with JD_SMALL_SEGMENT_HANDLING. In reality there is no connection between JD_SMALL_SEGMENT_HANDLING and fix of junction deviation. The only reason why I pulled JD_SMALL_SEGMENT_HANDLING from upstream is that I thought it might fix the problem observed. It fixed nothing so I analyzed the problem and found out, that the problem is caused by imprecise junction angle computation. The problem is there regardless JD_SMALL_SEGMENT_HANDLING  is enabled or disabled. I improved junction angle precision by using float source of position from gcode instead of position rounded to steps.

I also checked if the problem is still present in upstream. I have noticed the problem doesn't manifest itself upstream. Even though imprecise position is still used upstream. In upstream Marlin, the problem is somehow mitigated by always normalizing unit vector if there is any E move.

It might have sense to use float position if HAS_POSITION_FLOAT is enabled and fall back to original code if HAS_POSITION_FLOAT is disabled.

Reply:
Ok, I would check for HAS_POSITION_FLOAT then. Although we have several places that use/require POSITION_FLOAT, my point was just to not to require it outright.

Reply:
I have gated it by HAS_POSITION_FLOAT. And also melded in upstream workaround for case HAS_POSITION_FLOAT is unavailable.

Reply:
I have also removed redundant `&& DISABLED(CLASSIC_JERK)` as everything is inside DISABLED(CLASSIC_JERK) block.